prov/verbs/src/windows/verbs_nd_ibv.c

```diff
@@ -806,34 +812,61 @@ int ibv_post_send(struct ibv_qp *qp, struct ibv_send_wr *wr,
 		return errno;
 	}
 
-	// TODO: Add support for other transfer methods.
-	if (wr->opcode != IBV_WR_SEND) {
-		VRB_WARN(FI_LOG_EP_DATA, "Unsupported opcode: %d\n",
-			 wr->opcode);
-		assert(wr->opcode == IBV_WR_SEND);
-		errno = EINVAL;
-		return errno;
-	}
-
 	*bad_wr = NULL;
 
 	qp_nd = container_of(qp, struct nd_qp, qp);
 	while (wr) {
-		length = 0;
-		for (int i = 0; i < wr->num_sge; ++i) {
-			length += wr->sg_list[i].length;
-		}
-		flags = 0;
-		if (length <= qp_nd->max_inline_data) {
-			flags |= ND_OP_FLAG_INLINE;
+		switch (wr->opcode) {
+		case IBV_WR_SEND:
+			flags = 0;
+			flags |= check_for_inline(wr, qp_nd);
+			hr = qp_nd->nd2qp->lpVtbl->Send(qp_nd->nd2qp,
+							(void *)wr->wr_id,
+							(ND2_SGE *)wr->sg_list,
+							wr->num_sge, flags);
+			VRB_DBG(FI_LOG_EP_DATA,
+				"IND2QueuePair::Send: hr=0x%08lx, nsge=%d, addr=0x%016llx, length=%d, lkey=0x%08lx\n",
+				hr, wr->num_sge, wr->sg_list[0].addr,
+				wr->sg_list[0].length, wr->sg_list[0].lkey);
+			break;
+
+		case IBV_WR_RDMA_READ:
+			hr = qp_nd->nd2qp->lpVtbl->Read(
+				qp_nd->nd2qp, (void *)wr->wr_id,
+				(ND2_SGE *)wr->sg_list, wr->num_sge,
+				wr->wr.rdma.remote_addr, wr->wr.rdma.rkey,
+				flags);
+			VRB_DBG(FI_LOG_EP_DATA,
+				"IND2QueuePair::Read: hr=0x%08lx, nsge=%d, addr=0x%016llx, raddr=0x%016llx, rkey=0x%08lx, lkey=0x%08lx\n",
```

Comment:
line lengths are getting long

Reply:
Hmm, looks like clang-format doesn't like to split a string literal.